SwanGallery/src/handler.ts

```diff
@@ -0,0 +1,52 @@
+import { URLExt } from '@jupyterlab/coreutils';
+
+import { ServerConnection } from '@jupyterlab/services';
+
+/**
+ * Call the API extension
+ *
+ * @param endPoint API REST end point for the extension
+ * @param init Initial values for the request
+ * @returns The response body interpreted as JSON
+ */
+
+export async function requestAPI<T>(
+  projUrl: string,
+  endPoint = '',
+  init: RequestInit = {}
+): Promise<T> {
+  
+  let settings = ServerConnection.makeSettings();
+  
+  let requestUrl = URLExt.join(
+    settings.baseUrl,
+    'SwanGallery', // API Namespace
+    endPoint
+  );
+
+  requestUrl = requestUrl+"?url="+projUrl; //Add project url parameter 
+
+  let response: Response;
+  try {
+    response = await ServerConnection.makeRequest(requestUrl, init, settings);
+    }   
+    catch (error) {
+    throw new ServerConnection.NetworkError(<TypeError>error);
+  }
+
+  let data: any = await response.text();
```

Comment:
Can this be simplified with `response.json()` directly?

Also the `catch` in line 43 swallows the error returning the text, is that intentional?

Reply:
You are right, I simplified it and works fine. 

``` typescript
  let data: any;

  try {
    data = await response.json();
  } catch (error) {
    console.log('Not a JSON response body.', error);
  }
```
As for the error, it was giving me more details that way when I was testing. I will make it use `error` now.